### ISSUE TITLE
Upgrade to Kotlin 1.3.7x or 1.4x

### ISSUE THREAD

**Issue by rachelcarmena** [arrow-kt/arrow-meta#554]
It seems it's not possible because of all external symbols appear unbound in IR right now.

Tasks:

- [ ] Upgrade

- [ ] Enable the bot again (just checking compilation right now)

**Comment by rachelcarmena** [comment]
Last errors with 1.4x hided the new 1.3.72 Kotlin version. Let's try it!

**Comment by rachelcarmena** [comment]
`1.3.72` doesn't fix IR problem.

**Comment by iirekm** [comment]
+ 1

in particular I get:
```
java.lang.NoSuchMethodError: 'org.jetbrains.kotlin.com.intellij.openapi.extensions.impl.ExtensionPointImpl[] org.jetbrains.kotlin.com.intellij.openapi.extensions.impl.ExtensionsAreaImpl.getExtensionPoints()'
	at arrow.meta.internal.registry.InternalRegistry$DefaultImpls.registerMetaComponents(InternalRegistry.kt:132)
	at arrow.meta.Meta$DefaultImpls.registerMetaComponents(Meta.kt)
```

**Comment by rachelcarmena** [comment]
Pending features with Kotlin `1.4.10`:

* TypeClassesTest > `polymorphic constrain` - Issue: https://github.com/arrow-kt/arrow-meta/issues/739
* Kinds > `Platform types can be kinded ad-hoc by proof` - Issue: [pending]

It's necessary to continue reviewing these cases in the next versions.

**Comment by rachelcarmena** [comment]
Solved #706
